src/cmd/logic-function.js

```diff
@@ -18,123 +18,196 @@ const logicFunctionTemplatePath = path.join(__dirname, '/../../assets/logicFunct
 module.exports = class LogicFunctionsCommand extends CLICommandBase {
 	constructor(...args) {
 		super(...args);
+		this.api = createAPI();
+		this.logicFuncList = null;
+		this.org = null;
 	}
 
 	async list({ org }) {
-		const api = createAPI();
-		try {
-			const logicFunctions = await api.getLogicFunctionList({ org });
-			const orgName = getOrgName(org);
-			const list = logicFunctions.logic_functions;
-			if (list.length === 0) {
-				this.ui.stdout.write(`No Logic Functions currently deployed in your ${orgName}.`);
-				this.ui.stdout.write(`To create a Logic Function, see \`particle logic-function create\`.${os.EOL}
-							To download an existing Logic Function, see \`particle lf get\`.${os.EOL}`);
-			} else {
-				this.ui.stdout.write(`Logic Functions currently deployed in your ${orgName}:${os.EOL}`);
-				list.forEach((item) => {
-					// We assume atleast one trigger
-					this.ui.stdout.write(`- ${item.name} (${ item.enabled ? this.ui.chalk.cyanBright('enabled') : this.ui.chalk.grey('disabled') })${os.EOL}`);
-					this.ui.stdout.write(`	- ID: ${item.id}${os.EOL}`);
-					this.ui.stdout.write(`	- ${item.logic_triggers[0].type} based trigger ${os.EOL}`);
-				});
-				this.ui.stdout.write(this.ui.chalk.yellow(`${os.EOL}To view a Logic Function's code, see \`particle lf get.\`${os.EOL}`));
-				return list;
-			}
-			return list;
-		} catch (e) {
-			throw createAPIErrorResult({ error: e, message: 'Error listing logic functions' });
+		await this._setOrg(org);
+
+		await this._getLogicFunctionList();
+
+		if (this.logicFuncList === null || this.logicFuncList.length === 0) {
+			this._printListHelperOutput();
+		} else {
+			this._printListOutput({ logicFunctionsList: this.logicFuncList });
 		}
 	}
 
+	_printListHelperOutput() {
+		this.ui.stdout.write(`No Logic Functions currently deployed in your ${getOrgName(this.org)}.${os.EOL}`);
+		this.ui.stdout.write(`${os.EOL}`);
+		this.ui.stdout.write('To create a Logic Function, see  ' + this.ui.chalk.yellow('\'particle logic-function create\'.') + `${os.EOL}`);
+		this.ui.stdout.write('To download an existing Logic Function, see ' + this.ui.chalk.yellow('\'particle logic-function get\'.') + `${os.EOL}`);
+	}
+
+	_printListOutput({ logicFunctionsList }) {
+		this.ui.stdout.write(`Logic Functions currently deployed in your ${getOrgName(this.org)}:${os.EOL}`);
```

Comment:
General comment for another PR: make `getOrgName` return "your Sandbox" or the name of the org so that the messages print like this:
Logic Functions currently deployed in your Sandbox
Logic Functions currently deployed in Particle

instead of currently which would show
Logic Functions currently deployed in your Particle